### PR TITLE
Turn off jb eval

### PR DESCRIPTION
**Pull request by guillaumevernieres**
I'm not 100% sure this fixes the memory issue on MSU HPC, but, while I was working on other things, I turned off the jb eval out of impatience to gain ~5+mn of runtime. I also noticed that the memory footprint is down to ~0.8TB for the 3DVARFGAT.
I'm testing with more iterations right now.

### CODE REVIEW

File: 3dfgat.yaml.j2 [hunk 1]
```diff
@@ -5,6 +5,8 @@ cost function:
 
   cost type: 3D-FGAT
 
+  jb evaluation: false
```
Comment (danholdaway) [inline 3dfgat.yaml.j2]:
```suggestion
  jb evaluation: {{ jb_evaluation | default(true) }}
```

File: 3dvar.yaml.j2 [hunk 2]
```diff
@@ -5,6 +5,8 @@ cost function:
 
   cost type: 3D-Var
 
+  jb evaluation: false
```
Comment (danholdaway) [inline 3dvar.yaml.j2]:
```suggestion
  jb evaluation: {{ jb_evaluation | default(true) }}
```

File: 3dfgat.yaml.j2 [hunk 3]
```diff
@@ -118,3 +120,6 @@ test:
   float absolute tolerance: {{test_float_absolute_tolerance | default(0.0, true) }}
   integer tolerance: {{test_integer_tolerance | default(0, true) }}
 {% endif %}
+
+# Turn off the final J eval
+final j evaluation: false
```
Comment (danholdaway) [inline 3dfgat.yaml.j2]:
```suggestion
final j evaluation: {{ final_j_evaluation | default(true) }}
```

File: 3dvar.yaml.j2 [hunk 4]
```diff
@@ -112,3 +114,6 @@ test:
   float absolute tolerance: {{test_float_absolute_tolerance | default(0.0, true) }}
   integer tolerance: {{test_integer_tolerance | default(0, true) }}
 {% endif %}
+
+# Turn off the final J eval
+final j evaluation: false
```
Comment (danholdaway) [inline 3dvar.yaml.j2]:
```suggestion
final j evaluation: {{ final_j_evaluation | default(true) }}
```

**Review by danholdaway — CHANGES_REQUESTED** [review]
Thanks for adding this @guillaumevernieres. For backward compatibility I think it would be good to default to `True`. And also allow users the option to keep using the feature. Then you can set the two settings to `False` in your jcb settings.

**Comment by danholdaway** [comment]
By the way, when you create a jcb-algorithms PR, you also need to create an empty branch with the same name in the main jcb repo. This is unfortunately needed for the CI. It allows for the connection between jcb-algorithms and repos like jcb-gdas so that these changes can be tested against those repos. I'll try to remove that requirement soon.

**Comment by guillaumevernieres** [comment]
Roger that @danholdaway ... 

It looks like this is way more than a 5 mn gain. 
 - 30 iterations wo j eval : 17mn
 -  "          "         with         : probably 30+mn. I's still running but I'm pretty sure it will time out (only asked for 30mn on debug)
 
We should turn that option off by default, this has been a huge waste of resources.

**Comment by danholdaway** [comment]
I'm OK with setting the flags to e.g. `jb evaluation: {{ jb_evaluation | default(false) }}`

**Review by danholdaway — APPROVED** [review]

**Pull request merged**